commit 9a70aaafc4d8f7670197eec351ad6e5b20f6911e
Author: Matthias Klumpp <matthias@tenstral.net>
Date:   Mon May 26 22:41:27 2014 +0200

    Add AppStream metadata
    
    See http://techbase.kde.org/MetaInfo for more information about the
    metadata and how to create it.

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a362609..92929a2 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -56,7 +56,7 @@ install( TARGETS kgrapheditor ${INSTALL_TARGETS_DEFAULT_ARGS})
 install( FILES kgrapheditorsettings.kcfg DESTINATION ${KCFG_INSTALL_DIR} )
 install( FILES kgrapheditorui.rc DESTINATION ${DATA_INSTALL_DIR}/kgrapheditor )
 install( PROGRAMS kgrapheditor.desktop DESTINATION ${XDG_APPS_INSTALL_DIR} )
-
+install( FILES kgraphviewer.appdata.xml DESTINATION ${SHARE_INSTALL_PREFIX}/appdata/ )
 
 
 #add_subdirectory(plasmaapplet)
diff --git a/src/kgraphviewer.appdata.xml b/src/kgraphviewer.appdata.xml
new file mode 100644
index 0000000..e0a9594
--- /dev/null
+++ b/src/kgraphviewer.appdata.xml
@@ -0,0 +1,42 @@
+<?xml version="1.0" ?>
+<component type="desktop">
+  <id>kgraphviewer.desktop</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-2.0+</project_license>
+  <name>KGraphViewer</name>
+  <summary>A Graphviz dot graph viewer for KDE</summary>
+  <description>
+    <p>
+      KGraphViewer is a Graphviz dot graph file viewer, aimed to replace the other outdated GraphViz tools.
+      Graphs are commonly used in scientific domains and particularly in computer science.
+    </p>
+    <p>Features:</p>
+    <ul>
+      <li>Zooming</li>
+      <li>Threaded loading of several graphs in tabs</li>
+      <li>Saving of the recent files list</li>
+      <li>Manual reload of files</li>
+      <li>Display of a bird-eye view of the graph</li>
+      <li>Moving of the graph by dragging</li>
+      <li>Full featured printing</li>
+      <li>Perfect drawing of all graphviz example graphs</li>
+      <li>Automatically choose dot for directed graphs and neato for undirected</li>
+      <li>Possibility to use an arbitrary layout algo as soon as it produces xdot format</li>
+      <li>Automatic reloading with user confirmation of (externally) modified files (configurable)</li>
+      <li>Open new instances as new tabs in the old one (configurable)</li>
+    </ul>
+  </description>
+  <url type="homepage">http://extragear.kde.org/apps/kgraphviewer/</url>
+  <url type="bugtracker">https://bugs.kde.org/enter_bug.cgi?format=guided&amp;product=kgraphviewer</url>
+  <url type="help">http://docs.kde.org/development/en/extragear-graphics/kgraphviewer/index.html</url>
+  <screenshots>
+    <screenshot type="default">
+      <image>http://kde.org/images/screenshots/kgraphviewer.png</image>
+    </screenshot>
+  </screenshots>
+  <project_group>KDE</project_group>
+  <provides>
+    <binary>kgraphviewer</binary>
+    <binary>kgrapheditor</binary>
+  </provides>
+</component>
diff --git a/src/part/CMakeLists.txt b/src/part/CMakeLists.txt
index d27d1eb..041536d 100644
--- a/src/part/CMakeLists.txt
+++ b/src/part/CMakeLists.txt
@@ -16,7 +16,8 @@ kde4_add_library(kgraphviewerlib SHARED ${kgraphviewerlib_LIB_SRCS})
 add_definitions(-DQT_STL)
 target_link_libraries(kgraphviewerlib ${QT_QTSVG_LIBRARY} ${KDE4_KIO_LIBS} ${graphviz_LIBRARIES})
 
-set_target_properties(kgraphviewerlib PROPERTIES VERSION 2.1 SOVERSION 2 OUTPUT_NAME kgraphviewer )
+set(KGRAPHVIEWER_SOVERSION "2")
+set_target_properties(kgraphviewerlib PROPERTIES VERSION 2.1 SOVERSION ${KGRAPHVIEWER_SOVERSION} OUTPUT_NAME kgraphviewer )
 
 install( TARGETS kgraphviewerlib ${INSTALL_TARGETS_DEFAULT_ARGS})
 
@@ -32,6 +33,8 @@ target_link_libraries(kgraphviewerpart ${KDE4_KPARTS_LIBS} gvc cgraph pathplan c
 
 install( TARGETS kgraphviewerpart DESTINATION ${PLUGIN_INSTALL_DIR})
 
+########### configure AppStream metadata ###############
+configure_file(kgraphviewer.metainfo.xml.in "${CMAKE_CURRENT_BINARY_DIR}/kgraphviewer.metainfo.xml" IMMEDIATE @ONLY)
 
 ########### install files ###############
 
@@ -44,6 +47,7 @@ install( FILES chain-broken.png DESTINATION ${DATA_INSTALL_DIR}/kgraphviewerpart
 install( FILES kgraphviewer-bev.png DESTINATION ${DATA_INSTALL_DIR}/kgraphviewerpart/pics )
 install( FILES kgraphviewer-newnode.png DESTINATION ${DATA_INSTALL_DIR}/kgraphviewerpart/pics )
 install( FILES kgraphviewer-newedge.png DESTINATION ${DATA_INSTALL_DIR}/kgraphviewerpart/pics )
+install( FILES "${CMAKE_CURRENT_BINARY_DIR}/kgraphviewer.metainfo.xml" DESTINATION ${SHARE_INSTALL_PREFIX}/appdata/${CMAKE_LIBRARY_ARCHITECTURE}/ )
 
 install(FILES kgraphviewer_interface.h DESTINATION ${INCLUDE_INSTALL_DIR}/kgraphviewer/ COMPONENT Devel)
 install(FILES dotgraphview.h DESTINATION ${INCLUDE_INSTALL_DIR}/kgraphviewer/ COMPONENT Devel)
diff --git a/src/part/kgraphviewer.metainfo.xml.in b/src/part/kgraphviewer.metainfo.xml.in
new file mode 100644
index 0000000..970b1cb
--- /dev/null
+++ b/src/part/kgraphviewer.metainfo.xml.in
@@ -0,0 +1,14 @@
+<?xml version="1.0" ?>
+<component>
+  <id>libkgraphviewer</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-2.0+</project_license>
+  <name>KGraphViewer Library</name>
+  <summary>A Graphviz dot graph viewer for KDE (shared library)</summary>
+  <url type="homepage">http://extragear.kde.org/apps/kgraphviewer/</url>
+  <url type="bugtracker">https://bugs.kde.org/enter_bug.cgi?format=guided&amp;product=kgraphviewer</url>
+  <project_group>KDE</project_group>
+  <provides>
+    <library>kgraphviewer.so.@KGRAPHVIEWER_SOVERSION@</library>
+  </provides>
+</component>
